criu/uffd.c

```diff
@@ -269,6 +269,10 @@ int uffd_open(int flags, unsigned long *features)
 
 	uffd = syscall(SYS_userfaultfd, flags);
 	if (uffd == -1) {
+		if (flags == 0) {
```

Comment:
The usage of flags is a bad idea. Although it would work, somebody could add a new flag in kerndat and they would not suspect a side effect like this. Those flags are for the syscall and they should not be used for anything else, because people don't expect them to be. A new parameter would be a better idea in my opinion.